src/NuGet.Indexing/Handlers/SegmentToMainReaderMappingHandler.cs

```diff
@@ -0,0 +1,56 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Lucene.Net.Documents;
+using Lucene.Net.Index;
+using NuGet.Versioning;
+
+namespace NuGet.Indexing
+{
+    public class SegmentToMainReaderMappingHandler : IIndexReaderProcessorHandler
+    {
+        private readonly Dictionary<string, int[]> _mapping = new Dictionary<string, int[]>(StringComparer.Ordinal);
```

Comment:
Do you see a possibility to somehow keep the dictionary around? Per index reload we're recreating a few 100k's of variables, which is bound to induce a GC.